specification/dartLangSpec.tex

```diff
@@ -8848,53 +8851,84 @@ \subsubsection{Unqualified Invocation}
 }
 
 \LMHash{}%
-It is a compile-time error if $i$ occurs inside a top-level or static function
-(be it function, method, getter, or setter)
-or a top-level or static variable initializer,
-and there is no lexically visible declaration named \id{} in scope.
+For the static analysis of $i$,
+a lexical lookup of \id{} is performed
+(\ref{lexicalLookup}):
 
 \LMHash{}%
-If there exists a lexically visible declaration named \id,
-let $D_{id}$ be the innermost such declaration.
-Then:
+\Case{Lexical lookup yields a declaration}
+In this case the lexical lookup of \id{}
+yields a declaration $D$.
 \begin{itemize}
-\item Consider the situation where $D_{id}$ is a type declaration.
-  If $D_{id}$ is a declaration of a class $C$
+\item
+  Consider the case where $D$ is a type declaration.
```

Comment:
Just:

> If $D$ is a type declaration

(and say what a "type declaration" is. I assume it means a class declaration, mixin declaration or type alias declaration. Does it include a type variable?).

Reply:
I tried to get rid of 'Consider the case' but we need to make sure that it establishes a premise for the rest of this bullet point (it shouldn't be a reasonable interpretation that it's only valid for half of the bullet point). I think I found a way to preserve that property which is still less verbose.

Changed to use an explicit list of non-class type declarations at the end.